framework-test/src/main/java/org/checkerframework/framework/test/diagnostics/TestDiagnosticUtils.java

```diff
@@ -21,14 +21,26 @@
 public class TestDiagnosticUtils {
 
     /** How the diagnostics appear in Java source files. */
+    // (?s) enables the dot-all mode, which allows the dot to match any character, so it detects
+    // line separator \r\n in Windows.
     public static final String DIAGNOSTIC_IN_JAVA_REGEX =
-            "\\s*(error|fixable-error|warning|fixable-warning|other):\\s*(\\(?.*\\)?)\\s*";
+            "(?s)\\s*(error|fixable-error|warning|fixable-warning|other):\\s*(\\(?.*\\)?)\\s*";
 
     /** How the diagnostics appear in Java source files. */
     public static final Pattern DIAGNOSTIC_IN_JAVA_PATTERN =
             Pattern.compile(DIAGNOSTIC_IN_JAVA_REGEX);
 
-    public static final String DIAGNOSTIC_WARNING_IN_JAVA_REGEX = "\\s*warning:\\s*(.*\\s*.*)\\s*";
+    /**
+     * Regular expression pattern to match Java diagnostic warnings.
+     *
+     * <p>The pattern uses the dot-all mode (?s) to ensure that the dot (.) matches any character,
+     * including line terminators, which is necessary for detecting multi-line warning messages on
+     * both linux and Windows operating system.
+     */
+    public static final String DIAGNOSTIC_WARNING_IN_JAVA_REGEX =
```

Comment:
This is just the `warning` special case of the above, so I again don't see how DOTALL mode helps here.

Reply:
In the file `TestDiagnosticUtils.java`, the original `-1: other: `was triggered on line 196: [lineNo = -1](https://github.com/eisop/checker-framework/blob/4a69c27aa692ac0836998bcdfde76e3e306a1978/framework-test/src/main/java/org/checkerframework/framework/test/diagnostics/TestDiagnosticUtils.java#L195). I noticed that the token `-1: other: ` was followed directly by the expected error message. I thought it was the cause of the problem.

On [fromJavaxToolsDiagnostic](https://github.com/eisop/checker-framework/blob/4a69c27aa692ac0836998bcdfde76e3e306a1978/framework-test/src/main/java/org/checkerframework/framework/test/diagnostics/TestDiagnosticUtils.java#L105), it calls `fromPatternMatching` to Instantiate a diagnostic. At this point, the diagnostic message is already formatted with it's line number as the prefix:
```
fromJavaFileComment:         error: (assignment.type.incompatible)
fromJavaxToolsDiagnostic:    5:  error: (assignment.type.incompatible)
```

The original `DIAGNOSTIC_WARNING_IN_JAVA_REGEX` does not detect such prefix; so it sends the diagnostic message to the line 196 I mentioned above, and therefore produces the mismatched message. When `DIAGNOSTIC_WARNING_IN_JAVA_REGEX` detects the prefix, the message will be handled by [line 161](https://github.com/eisop/checker-framework/blob/4a69c27aa692ac0836998bcdfde76e3e306a1978/framework-test/src/main/java/org/checkerframework/framework/test/diagnostics/TestDiagnosticUtils.java#L161). And in this case, the line number prefix is properly parsed. Since `fromJavaxDiagnosticList` detects equal error message, only one correct diagnostic is printed. 

In addition, you are right, I noticed that the DotAll is not necessary for the warning regex, I think that is because all the diagnostic messages here are errors. So in this case `DIAGNOSTIC_WARNING_IN_JAVA_REGEX` is not used. 

I think the match checked by [line 272](https://github.com/eisop/checker-framework/blob/4a69c27aa692ac0836998bcdfde76e3e306a1978/framework-test/src/main/java/org/checkerframework/framework/test/diagnostics/TestDiagnosticUtils.java#L272) is not the same type of the error message here.